### PR TITLE
Feature/non obligated weee received report

### PR DESCRIPTION
**Pull request by willdot**
PBI: 73021

Note: The tests for getting active compliance years, can be found on Stephens branch.

### CODE REVIEW

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Reports/GetUkNonObligatedWeeeReceivedDataCsvHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
+﻿namespace EA.Weee.RequestHandlers.Tests.Unit.Admin.Reports
+{
+    using System;
+    using System.Security;
+    using System.Threading.Tasks;
+    using Core.Admin;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Reports/GetUkNonObligatedWeeeReceivedDataCsvHandlerTests.cs]: Can some tests be added to check the CSV, similar to that in GetAatfAeReturnDataCsvHandlerTests

File: src/EA.Weee.DataAccess/StoredProcedure/StoredProcedures.cs [hunk 2]
```diff
@@ -244,6 +244,15 @@ public async Task<List<UkEeeCsvData>> SpgUKEEEDataByComplianceYear(int complianc
                     complianceYearParameter).ToListAsync();
         }
 
+        public async Task<List<UkNonObligatedWeeeReceivedData>> UKUkNonObligatedWeeeReceivedByComplianceYear(int complianceYear)
+        {
+            var complianceYearParameter = new SqlParameter("@ComplianceYear", complianceYear);
+
+            // Put the correct stored procedure name here
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.DataAccess/StoredProcedure/StoredProcedures.cs]: Guess don't need this comment?

File: src/EA.Weee.RequestHandlers/Admin/GetActiveComplianceYears/GetAatfReturnsActiveComplianceYearsHandler.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        public GetAatfReturnsActiveComplianceYearsHandler(IWeeeAuthorization authorization, IGetAatfReturnsActiveComplianceYearsDataAccess dataAccess)
+        {
+            this.authorization = authorization;
+            this.dataAccess = dataAccess;
+        }
+
+        public async Task<List<int>> HandleAsync(GetAatfReturnsActiveComplianceYears message)
+        {
+            authorization.EnsureCanAccessInternalArea();
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Admin/GetActiveComplianceYears/GetAatfReturnsActiveComplianceYearsHandler.cs]: Add Tests, for the auto and the data access call

File: src/EA.Weee.RequestHandlers/Admin/GetActiveComplianceYears/GetAatfReturnsActiveComplianceYearsDataAccess.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        private readonly WeeeContext context;
+
+        public GetAatfReturnsActiveComplianceYearsDataAccess(WeeeContext context)
+        {
+            this.context = context;
+        }
+
+        public async Task<List<int>> Get()
+        {
+            return await context.Returns
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Admin/GetActiveComplianceYears/GetAatfReturnsActiveComplianceYearsDataAccess.cs]: Add data access tests and tests to ensure the return data is ordered correctly

File: src/EA.Weee.RequestHandlers/Admin/GetActiveComplianceYears/GetAatfReturnsActiveComplianceYearsHandler.cs (hunk 3 above)
Comment (willdot) [reply src/EA.Weee.RequestHandlers/Admin/GetActiveComplianceYears/GetAatfReturnsActiveComplianceYearsHandler.cs]: Stephen has the tests for this code on his branch.

File: src/EA.Weee.RequestHandlers/Admin/GetActiveComplianceYears/GetAatfReturnsActiveComplianceYearsDataAccess.cs (hunk 4 above)
Comment (willdot) [reply src/EA.Weee.RequestHandlers/Admin/GetActiveComplianceYears/GetAatfReturnsActiveComplianceYearsDataAccess.cs]: Stephen has the tests for this code on his branch.

**Comment by willdot** [comment]
> The

Removed the Using and deleting temp tables now.

The AC says the the year rows will include totals for all quarters for the category. Can't find the example report, but I remember the years being in the same column at the bottom.

Yh the parent id check is looking for the most recent submission and ignoring any edited submissions.

**Pull request merged**